engines/dfc_provider/spec/system/orders_backorder_spec.rb

```diff
@@ -0,0 +1,99 @@
+
+  # Distributor has imported a copy of the product and is selling it
+  # The distributor must be authorised with the email address of the supplier owner.
+  # Because they are on the same OFN instance in this test, they must be the same user.
+  let(:distributor_owner) { supplier_owner }
+  let(:distributor) {
+    create(:distributor_enterprise, id: 11_000, owner: distributor_owner, name: "Shane's Shop")
+  }
+  let(:distributor_variant) {
+    create(:variant, id: 11_001, unit_value: 1, sku: "DIST", on_hand: 10,
+                     supplier: distributor).tap { |v|
+      v.semantic_links.create(semantic_id:) # variant is linked to supplier variant
+      v.on_demand = false
+    }
+  }
+  let(:distributor_order_cycle) {
+    create(:simple_order_cycle, distributors: [distributor], variants: [distributor_variant])
+  }
+
+  # Customer orders the distributed product
+  let(:order) {
+    create(:order_with_totals_and_distribution,
+           :with_line_item,
+           variant: distributor_variant,
+           distributor:, order_cycle: distributor_order_cycle)
+  }
+
+  before {
+    # For this test, OFN is talking to itself. We currently don't allow that IRL.
+    allow(PrivateAddressCheck).to receive(:private_address?).and_return(false)
+
+    # Pretend that user has authenticated with OIDC
+    distributor_owner.oidc_account.update!(token: allow_token_for(email: distributor_owner.email))
+
+    variant.on_hand = 1
+  }
+
+  xit "debugging: accesses the webserver" do
+    # try accessing the supplier's catalog
+    url = "http://#{host}/api/dfc/enterprises/#{supplier.id}/catalog_items"
+
+    DfcRequest.new(distributor_owner).call(url)
+    result = DfcRequest.new(distributor_owner).call(url) # make a second http request, also works.
+    DfcIo.import(result)
+
+    puts "\nRequest: #{url}\n"
+    puts "Response includes:"
+    puts result.scan(/[^"]*10001[^"]*/)
+    # http://127.0.0.1:56286/api/dfc/enterprises/10000/catalog_items/10001
+    # http://127.0.0.1:56286/api/dfc/enterprises/10000/supplied_products/10001
+  end
+
+  describe "BackorderJob" do
+    it "creates an order for the source variant" do
+      # For debugging you can check log/test.log, eg:
+      # echo "" > log/test.log; tail -f log/test.log | egrep "(Started|Completed)"
+
+      expect {
+        # BackorderJob.perform_now(order) # Cannot perform whole job as it gets stuck on record lock
+        BackorderJob.new.place_backorder(order)
+      }.to change { supplier.distributed_orders.count }.by(1)
+
+      order = supplier.distributed_orders.first
+      expect(order.created_by).to eq distributor_owner
+      expect(order.user).to eq distributor_owner
+      expect(order.email).to eq distributor_owner.email
+      expect(order.state).to eq "complete"
+
+      expect(order.line_items.count).to eq 1
+      expect(order.line_items.first.variant).to eq variant
+      expect(order.line_items.first.quantity).to eq 1
+    rescue Faraday::UnprocessableEntityError => e
```

Comment:
This should have a different name, `order` is already defined in a `let` . If I am following correctly this is the order created by the DFC endpoint, so maybe we could call it `dfc_created_order` ?

Reply:
Good point, thanks. I opted to rename to supplier_order and distributor_order so it's clear which enterprise they are linked to.